File: chapt2/subquery.js
let query = 'SELECT * FROM songs';
if (req.params.artist) {
    query = query + ' WHERE artist = "' + req.params.artist + '"'
}
query = query + ' LIMIT 10';

connection.query(query, (err, data) => {
    if (err) throw err;
    res.render('index', { songs: data })
})

const select = (columns, table_name) => `SELECT ${columns} FROM ${table_name}`

const where = (query, column, match) => {
    if (column && match) return `${query} WHERE ${column} = ${match}`
    return query
}
const limit = (query, number) => {
    if (number) return `${query} LIMIT ${number}`
    return query
}

let query = limit(where(select(), "artist", req.params.artist), 10)

// or using experiment pipeline operator that reads much better
let query = select()
    |> where("artist", req.params.artist)
    |> limit(10)

File: chapt2/subquery.ex
def select(columns, table_name) do
  "SELECT #{columns} FROM #{table_name}"
end

def where(query, _column, nil), do: query
def where(query, column, match) do
  "#{query} WHERE #{column} = #{match}"
end

def limit(query, nil), do: query
def limit(query, number) do
  "#{query} LIMIT #{number}"
end

query = "*"
|> select("songs")
|> where("artist", req.params.artist)
|> limit(10)
